plugins/module_utils/foreman_helper.py

```diff
@@ -364,6 +462,34 @@ def sanitize_entity_dict(entity_dict, name_map):
     return {value: entity_dict[key] for key, value in name_map.items() if key in entity_dict}
 
 
+def _flatten_entity(entity, entity_spec):
+    """Flatten entity according to spec"""
+    result = {}
+    for key, value in entity.items():
+        if key in entity_spec:
+            spec = entity_spec[key]
+            _type = spec.get('type', 'str')
+            if _type == 'entity':
```

Comment:
I think having
```python
flat_name = spec.get('flat_name', key)
```
here and use flat_name in all accesses below would be more readable

Reply:
I'll try that.